src/rt/runtime.rs

```diff
@@ -140,23 +217,34 @@ impl Runtime {
 struct Machine {
     /// Holds the processor until it gets stolen.
     processor: Spinlock<Processor>,
+
+    id: usize,
+    drained: AtomicBool,
 }
 
 impl Machine {
     /// Creates a new machine running a processor.
-    fn new(p: Processor) -> Machine {
+    fn new(id: usize, p: Processor) -> Machine {
         Machine {
             processor: Spinlock::new(p),
+            id,
+            drained: AtomicBool::new(false),
         }
     }
 
     /// Schedules a task onto the machine.
-    fn schedule(&self, rt: &Runtime, task: Runnable) {
-        self.processor.lock().schedule(rt, task);
+    fn schedule(&self, rt: &Runtime, task: Task) {
+        if !self.drained.load(Ordering::SeqCst /* ??? */) {
```

Comment:
this is hot path, and I don't understand what SeqCst mean, maybe other values give better performance